modules/autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/credential/management/AutofillManagementActivity.kt

```diff
@@ -60,19 +62,30 @@ class AutofillManagementActivity : DuckDuckGoActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
+        observeViewModel()
+        viewModel.launchDeviceAuth(savedInstanceState)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        viewModel.lock()
+    }
 
-        // TODO implement reauth when app is backgrounded
-        deviceAuthenticator.authenticate(AUTOFILL, this) {
-            if (it == AuthResult.Success) {
-                if (savedInstanceState == null) {
+    override fun onWindowFocusChanged(hasFocus: Boolean) {
+        super.onWindowFocusChanged(hasFocus)
+        if (!hasFocus) viewModel.lock()
+    }
+
+    private fun launchDeviceAuth(savedInstanceState: Bundle?) {
+        if (deviceAuthenticator.hasValidDeviceAuthentication()) {
+            deviceAuthenticator.authenticate(AUTOFILL, this) {
+                if (it == AuthResult.Success) {
```

Comment:
I think this needs the else case to be handled to avoid the blank screen issue. maybe? 👇

```suggestion
                if (it == AuthResult.Success) {
                    viewModel.unlock()
                    showViewMode()
                } else {
                    viewModel.lock()
                }
```